### PR TITLE
Fix the expired links caused by the directory name change in optuna-examples

### LINKED ISSUE #5622: Fix the expired links caused by the directory name change in `optuna-examples`
### What is an issue?

The directory structure of optuna-examples related to basic and faq has changed in several pull requests, such as  https://github.com/optuna/optuna-examples/pull/277, causing some links to expire. The expired link is:
- https://github.com/optuna/optuna/blob/master/optuna/pruners/_successive_halving.py#L23

### PR DESCRIPTION
**Pull request by kAIto47802**
## Motivation

This PR aims to resolve the following issue:
- https://github.com/optuna/optuna/issues/5622

## Description of the changes

- Resolve #5622
- Fix the expired link to `basic/pruning.py` in `optuna-examples`

This PR depends on:
- https://github.com/optuna/optuna-examples/pull/277

As the PR above adds `basic/pruning.py`, this PR must be merged after [this PR](https://github.com/optuna/optuna-examples/pull/277).

### CODE REVIEW

**Review by eukaryo — COMMENTED** [review]
https://github.com/optuna/optuna-examples/blob/main/basic/pruning.py seems 404.

**Comment by eukaryo** [comment]
@y0z Could you review this PR?

**Comment by kAIto47802** [comment]
Thank you for your review. The PR https://github.com/optuna/optuna-examples/pull/277 has been merged, the link is now valid.

File: optuna/pruners/_successive_halving.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class SuccessiveHalvingPruner(BasePruner):
     Note that, this class does not take care of the parameter for the maximum
     resource, referred to as :math:`R` in the paper. The maximum resource allocated to a trial is
     typically limited inside the objective function (e.g., ``step`` number in `simple_pruning.py
-    <https://github.com/optuna/optuna-examples/blob/main/simple_pruning.py>`__,
+    <https://github.com/optuna/optuna-examples/blob/main/basic/pruning.py>`__,
```
[the remark this answers is not in the document]
Comment (kAIto47802) [reply optuna/pruners/_successive_halving.py:23]: @eukaryo @HideakiImamura 
Thank you for your suggestion. 
[The PR](https://github.com/optuna/optuna-examples/pull/277) in optuna-example changed the link from
https://github.com/optuna/optuna-examples/blob/main/basic_and_faq_usages/simple_pruning.py
to
https://github.com/optuna/optuna-examples/blob/main/basic/pruning.py .

I should have add description about this. Could you take another look?

**Comment by eukaryo** [comment]
Thank you for updating the description! I understand. I have confirmed https://github.com/optuna/optuna-examples/blob/main/basic/pruning.py is available now.

**Review by eukaryo — APPROVED** [review]
LGTM

**Review by HideakiImamura — APPROVED** [review]

**Pull request merged**